src/couch_prometheus/src/couch_prometheus_util.erl

```diff
@@ -167,3 +167,75 @@ val(Data) ->
 val(Key, Stats) ->
     {Key, Data} = lists:keyfind(Key, 1, Stats),
     val(Data).
+
+-ifdef(TEST).
+-include_lib("couch/include/couch_eunit.hrl").
+
+to_prom_test_() ->
+    [
+        ?_assertEqual(
+            <<"couchdb_ddoc_cache 10">>,
+            test_to_prom_output(ddoc_cache, counter, 10)
+        ),
+        ?_assertEqual(
```

Comment:
The tests setup looks unnecessarily complicated -- since we don't do any setup or teardown, they can be simpler.

```erlang
to_prom_ddoc_cache_test() ->
   ?assertEqual(<<"couchdb_ddoc_cache 10">>, prom_out(ddoc_cache, counter, 10)).

to_prom_gauge_test() ->
   ?assertEqual(...)
```

Or grouped but without the `test_` (generator format) as simple assert statements:

```erlang
to_prom_test() ->
      ?assertEqual(<<"couchdb_ddoc_cache 10">>, prom_out(ddoc_cache, counter, 10)),
      ?assertEqual(<<"couchdb_temperature_celsius 36">>, prom_out(temperature_celsius, gauge, 36)),
      ...
```

Reply:
@nickva this was just ported verbatim from the deleted [couch_prometheus_util.erl](s[rc/couch_prometheus/test/eunit/couch_prometheus_util_tests.erl](https://github.com/apache/couchdb/pull/4474/files#diff-693448a0247c2a0cf45a63e30a4f07d10099cb25a0fff201210c12ad25dee273) though happy to clean it up so long as it doesn't confuse things. 

The second example you suggest doesn't appear to work - the test runs (with a compiler warning `Warning: a term is constructed, but never used`), but the first assertion is ignored. The examples in [the documentation](https://www.erlang.org/doc/apps/eunit/chapter.html#an-example), look like they use a generator so am I missing something?

Reply:
I pushed an updated which is a hybrid of the two approaches, preserving the multiple assertions with a generator.

Reply:
thanks @nickva  - I'd missed the `_` prefix in the assertion. Pushed an update without generators.